server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotsStatusResponse.java

```diff
@@ -94,4 +87,15 @@ public boolean equals(Object o) {
     public int hashCode() {
         return snapshots != null ? snapshots.hashCode() : 0;
     }
+
+    @Override
+    public Iterator<? extends ToXContent> toXContentChunked() {
+        return Iterators.concat(
+            Iterators.single((ToXContent) (b, p) -> b.startObject().startArray("snapshots")),
+            snapshots.stream()
+                .flatMap(s -> StreamSupport.stream(Spliterators.spliteratorUnknownSize(s.toXContentChunked(), Spliterator.ORDERED), false))
+                .iterator(),
```

Comment:
optional: this seems a pretty useful pattern, maybe we should extract a utility for it now. Kinda surprised this is the first time it appears tbh.

Reply:
We have some variations of this here and there. I have a plan for this already in the making :) PR incoming.